### PR TITLE
Remove references to the string 'key' and change them to 'id'

### PR DESCRIPTION
**Pull request by nagilson**
VS Code has a filter for the string 'key' -- this causes our telemetry to be dropped when referencing the term 'install key' as it thinks it may be a secret. https://github.com/microsoft/vscode/blob/48ba01b5896376718aa66a11c60305cb105815c0/src/vs/platform/telemetry/common/telemetryUtils.ts#L336

Id is safe. I talked with the vscode team and there is no good alternative besides turning off all telemetry safety scanning, which I didn't want to do.

Hence, 'key' --> 'id'

### CODE REVIEW

File: vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts [hunk 1]
```diff
@@ -124,57 +124,57 @@ export class InstallTrackerSingleton
 
     /**
      *
-     * @param key the install key to get a working install promise for.
+     * @param id the install id to get a working install promise for.
      */
-    public getPromise(key : DotnetInstall) : Promise<string> | null
+    public getPromise(id : DotnetInstall) : Promise<string> | null
```
Comment (smitpatel) [inline vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts:129]: same better parameter name

File: vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts [hunk 2]
```diff
@@ -226,21 +226,21 @@ ${convertedInstalls.map(x => `${JSON.stringify(x.dotnetInstall)} owned by ${x.in
...
+    protected async removeVersionFromExtensionState(context : IAcquisitionWorkerContext, idStr: string, installIdObj: DotnetInstall)
     {
-        return this.executeWithLock( false, async (key: string, install: DotnetInstall) =>
+        return this.executeWithLock( false, async (id: string, install: DotnetInstall) =>
         {
-            this.eventStream.post(new RemovingVersionFromExtensionState(`Removing ${JSON.stringify(install)} with key ${key} from the state.`));
-            const existingInstalls = await this.getExistingInstalls(key === this.installedVersionsKey, true);
+            this.eventStream.post(new RemovingVersionFromExtensionState(`Removing ${JSON.stringify(install)} with id ${id} from the state.`));
+            const existingInstalls = await this.getExistingInstalls(id === this.installedVersionsid, true);
             const installRecord = existingInstalls.filter(x => IsEquivalentInstallation(x.dotnetInstall, install));
```
Comment (smitpatel) [inline vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts:244]: installedVersionsId

File: vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts [hunk 3]
```diff
@@ -315,38 +315,38 @@ ${convertedInstalls.map(x => `${JSON.stringify(x.dotnetInstall)} owned by ${x.in
...
     }
 
-    public async checkForUnrecordedLocalSDKSuccessfulInstall(context : IAcquisitionWorkerContext, dotnetInstallDirectory: string, installedInstallKeysList: InstallRecord[]): Promise<InstallRecord[]>
+    public async checkForUnrecordedLocalSDKSuccessfulInstall(context : IAcquisitionWorkerContext, dotnetInstallDirectory: string, installedInstallIdsList: InstallRecord[]): Promise<InstallRecord[]>
     {
-        return this.executeWithLock( false, async (dotnetInstallDir: string, installedInstallKeys: InstallRecord[]) =>
+        return this.executeWithLock( false, async (dotnetInstallDir: string, installedInstallIds: InstallRecord[]) =>
         {
-            let localSDKDirectoryKeyIter = '';
+            let localSDKDirectoryidIter = '';
```
Comment (smitpatel) [inline vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts:328]: LocalSDKDirectoryIdIter

**Review by smitpatel — APPROVED** [review]
Since this touch telemetry, also check if you need to classify the new property.

File: vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts (hunk 3 above)
Comment (nagilson) [reply vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts]: Thanks for your approval, looks like I forgot to do a case-sensitive search on this file. I also improved some of the names in other spots where it looks like I missed it.

You're right in that I will need to go into the GDPR tool and reclassify the events and update some of our queries as well.

**Review by smitpatel — APPROVED** [review]

**Pull request merged**